### PR TITLE
test(treeNode): test onToggle fn on treeNode

### PR DESCRIPTION
**Pull request by maximilianoforlenza**
test(treebeard): add more unit testing

### CODE REVIEW

File: src/components/common/index.js [hunk 1]
```diff
@@ -4,4 +4,9 @@ const Div = styled('div', {
     shouldForwardProp: prop => ['className', 'children'].indexOf(prop) !== -1
 })((({style}) => style));
 
+const Ul = styled('ul', {
```
Comment (navarroaxel) [inline src/components/common/index.js]: why you don't `export` on declaration?
```javascript
export const Ul = styled('ul', {
```

File: src/components/TreeNode/index.js [hunk 2]
```diff
@@ -104,11 +92,9 @@ class TreeNode extends PureComponent {
         const {node, style} = this.props;
         const decorators = this.decorators();
         const animations = this.animations();
-
         return (
-            <Li ref={this.topLevelRef}
-                style={style.base}>
-                <NodeHeader {...{decorators, animations, node, style}} onClick={() => this.onClick()}/>
+            <Li style={style.base}>
+                <NodeHeader id="button" {...{decorators, animations, node, style}} onClick={() => this.onClick()}/>
```
Comment (navarroaxel) [inline src/components/TreeNode/index.js]: the `id` should be unique to avoid issues with the HTML, the user of this package can get issues with this ID.

File: __tests__/Treebeard.js [hunk 3]
```diff
@@ -1,24 +1,13 @@
 import React from 'react';
```
Comment (navarroaxel) [inline __tests__/Treebeard.js]: why you don't use a `<component>.test.js` or `<component>.specs.js` in the filename?

File: __tests__/Treebeard.js [hunk 4]
```diff
@@ -1,24 +1,13 @@
...
-        ]
-    }]
-};
+import data from '../example/data';
 
 describe('<Treebeard/>', () => {
-    const wrapper = shallow(<Treebeard data={data}/>);
-    test('should exist ul', () => {
-        expect(wrapper.find('ul').exists()).toBe(true);
+    test('should render correctly treebeard component with props', () => {
```
Comment (navarroaxel) [inline __tests__/Treebeard.js]: the test description could be
```javascript
test(`should match snapshot`)
```
the `props` is a context situation, e.g.:
```javascript
describe(`when data is not null`, () => {
  test(`should match snapshot`, ...);
});
```

File: __tests__/Treebeard.js [hunk 5]
```diff
@@ -1,24 +1,13 @@
...
+        const wrapper = shallow(<Treebeard data={castArray(data)}/>);
+        expect(wrapper).toMatchSnapshot();
     });
```
Comment (navarroaxel) [inline __tests__/Treebeard.js]: you can add another snapshot test when `data is null`

File: __tests__/TreeNode.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            const clickFn = jest.fn(node => node);
+            const wrapper = shallow(<TreeNode
+                node={data}
+                decorators={defaultDecorators}
+                animations={defaultAnimations}
+                style={defaultTheme}
+                onToggle={clickFn}
+            />);
+            wrapper.find('#button').simulate('click');
+            expect(clickFn.mock.calls[0][0]).toEqual(data);
```
Comment (navarroaxel) [inline __tests__/TreeNode.js]: could be replaced with `expect(clickFn).toBeCalledWith(data)`?

File: __tests__/TreeNode.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import TreeNode from '../src/components/TreeNode';
+import defaultTheme from '../src/themes/default';
+import defaultAnimations from '../src/themes/animations';
+import defaultDecorators from '../src/components/Decorators';
+import data from '../example/data';
+
+describe('<TreeNode/>', () => {
+    describe('when onToggle function is called', () => {
+        test('should call correctly', () => {
+            const clickFn = jest.fn();
```
Comment (navarroaxel) [inline __tests__/TreeNode.js]: this function mock can be global on this file.

File: __tests__/TreeNode.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+            const wrapper = shallow(<TreeNode
```
Comment (navarroaxel) [inline __tests__/TreeNode.js]: you can create a function on this test file called `renderComponent` to render with default props and overwrite when you need.
```javascript
const renderComponent = (props = {}) => <TreeNode
  {...{
    decorators={defaultDecorators}
    ...props,
  }}
/>;
```
usage:
```javascript
const wrapper = shallow(renderComponent());
...
const wrapper = shallow(renderComponent({node: customNode}));
```

File: __tests__/TreeNode.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+                node={data}
+                decorators={defaultDecorators}
+                animations={defaultAnimations}
+                style={defaultTheme}
+                onToggle={clickFn}
+            />);
+            wrapper.find('#button').simulate('click');
```
Comment (navarroaxel) [inline __tests__/TreeNode.js]: you can declare a function to avoid the hardcoded id be copy-pasted several times on this file:
```javascript
const pageObjectModel = wrapper => ({
  button: () => wrapper.find('#button')
});
```

File: src/components/TreeNode/index.js [hunk 10]
```diff
@@ -1,32 +1,22 @@
...
 
 const Li = styled('li', {
     shouldForwardProp: prop => ['className', 'children', 'ref'].indexOf(prop) !== -1
 })(({style}) => style);
-const Ul = styled('ul', {
-    shouldForwardProp: prop => ['className', 'children', 'ref'].indexOf(prop) !== -1
-})(({style}) => style);
 
 class TreeNode extends PureComponent {
-    constructor(props) {
```
Comment (navarroaxel) [inline src/components/TreeNode/index.js]: you are merging new unit tests and refactor on the same commit.

**Comment by navarroaxel** [comment]
the commit message could be better
```
test(treebeard): add unit tests for TreeNode and Treebeard
```
